### PR TITLE
[PRM-855] - Inset text for duplicate uploads (JS)

### PR DESCRIPTION
**Pull request by LewisMitchell**
multi-duplicates
4-duplicates
3-duplicates
2-duplicates
1-duplicate

### CODE REVIEW

File: app/viewtils/EvidenceFileUploadsHelper.scala [hunk 1]
```diff
@@ -60,4 +65,31 @@ class EvidenceFileUploadsHelper @Inject()(govukInsetText: GovukInsetText,
...
+        uploads => {
+          val uploadedFiles: Seq[(UploadJourney, Int)] = uploads.zipWithIndex
+          val mappedDuplicates: Map[Option[String], Seq[(UploadJourney, Int)]] = uploadedFiles.groupBy(_._1.uploadDetails.map(_.checksum))
+
+          if (!mappedDuplicates.exists(_._2.size > 1)) {
+            None
+          } else if (mappedDuplicates.count(_._2.size > 1) > 1) {
+           Some(messages("otherReason.uploadList.multipleDuplicateInsetText", mappedDuplicates.map(_._2.size).sum))
+          } else {
+            val duplicates: Seq[(UploadJourney, Int)] = mappedDuplicates.filter(_._2.size > 1).head._2
```
Comment (ruthdas03) [inline app/viewtils/EvidenceFileUploadsHelper.scala]: Lines 81 to 88, written below to find the inset text is same as the lines 55 to 62 in the method 'duplicateDocumentInset'. Can these be refactored.

**Review by ruthdas03 — CHANGES_REQUESTED** [review]

**Comment by LewisMitchell** [comment]
I think you may have reviewed an older version of this branch @ruthdas03 ?

File: app/viewtils/EvidenceFileUploadsHelper.scala [hunk 2]
```diff
@@ -41,56 +40,38 @@ class EvidenceFileUploadsHelper @Inject()(govukInsetText: GovukInsetText,
...
+    ))
+  }
 
-    val mappedDuplicates = uploadedFiles.groupBy(_._1.uploadDetails.map(_.checksum))
+  def getInsetTextForUploads(uploads: Seq[(UploadJourney, Int)])(implicit messages: Messages): Option[Html] = {
+    getInsetTextMessage(uploads).fold[Option[Html]](None)(insetText => Some(govukInsetText(InsetText(content = Text(insetText)))))
+  }
 
+  def getInsetTextMessage(uploadedFiles: Seq[(UploadJourney, Int)])(implicit messages: Messages): Option[String] = {
+    val mappedDuplicates: Map[Option[String], Seq[(UploadJourney, Int)]] = uploadedFiles.groupBy(_._1.uploadDetails.map(_.checksum))
```
Comment (ruthdas03) [inline app/viewtils/EvidenceFileUploadsHelper.scala:57]: The boolean (_._2.size > 1), is being evaluated in line 58, 60 and 61 which looks redundant. Can it be stored in a boolean val instead.

**Review by ruthdas03 — COMMENTED** [review]
Looks ok, just a minor code optimisation, if you'd agree.

File: app/viewtils/EvidenceFileUploadsHelper.scala (hunk 2 above)
Comment (LewisMitchell) [reply app/viewtils/EvidenceFileUploadsHelper.scala:57]: Good idea

**Review by ruthdas03 — APPROVED** [review]

**Pull request merged**